frontend/src/components/columns/index.tsx

```diff
@@ -1,58 +1,80 @@
 import React from "react";
 
 import { Columns } from "fannypack";
-import styled, { css } from "styled-components";
-import { CustomColumnSideType, CustomColumnsType } from "./types";
+import styled, { css, CSSObject, SimpleInterpolation } from "styled-components";
 
-export const BaseCustomColumns: React.SFC<CustomColumnsType> = Columns;
+export type CustomColumnSideType = { desktop: number; mobile: number };
+export type CustomColumnSidesType = {
+  top?: CustomColumnSideType;
+  bottom?: CustomColumnSideType;
+  left?: CustomColumnSideType;
+  right?: CustomColumnSideType;
+};
 
-const renderResponsiveClasses = (
-  prop: {
-    top?: CustomColumnSideType;
-    bottom?: CustomColumnSideType;
-    left?: CustomColumnSideType;
-    right?: CustomColumnSideType;
-  },
-  property: "padding" | "margin",
-) => {
-  if (prop) {
-    let assembledStyle = ``;
-    for (const [key, element] of Object.entries(prop)) {
-      if (element) {
-        assembledStyle = `
-          ${assembledStyle}
-          ${property}-${key}: ${element.mobile}rem;
-          @media only screen and (min-width: 1024px) {
-            ${property}-${key}: ${element.desktop}rem;
-          }
-        `;
+const camelToKebabCase = (str: string) =>
+  str.replace(/[A-Z]/g, letter => `-${letter.toLowerCase()}`);
+
+type ResponsiveValue = {
+  desktop: number;
+  mobile: number;
+};
+
+export const media = {
+  desktop: (
+    first: CSSObject | TemplateStringsArray,
+    ...interpolations: SimpleInterpolation[]
+  ) => {
+    return css`
+      @media (min-width: 1024px) {
+        ${css(first, ...interpolations)}
       }
-    }
-    return `
-      ${assembledStyle}
     `;
-  }
+  },
+};
+
+const responsiveStyle = (
+  cssProp: keyof React.CSSProperties,
+  values: ResponsiveValue,
+) => {
+  const prop = camelToKebabCase(cssProp);
+
+  console.log(prop);
+  return css`
+    ${prop}: ${values.mobile}rem;
 
-  return css``;
+    ${media.desktop`
+      ${prop}: ${values.desktop}rem;
+    `}
+  `;
 };
 
+export type ResponsiveProps = {
+  marginTop?: ResponsiveValue;
+  marginBottom?: ResponsiveValue;
+  marginLeft?: ResponsiveValue;
+  marginRight?: ResponsiveValue;
+  paddingTop?: ResponsiveValue;
+  paddingBottom?: ResponsiveValue;
+  paddingLeft?: ResponsiveValue;
+  paddingRight?: ResponsiveValue;
+};
+
+export const responsiveSpacing = (props: ResponsiveProps) => css`
+  ${props.marginTop && responsiveStyle("marginTop", props.marginTop)}
```

Comment:
@patrick91 probably we can use the prop property name instead of the hardcoded name (eg: "marginTop")